sdk/eventgrid/azure-messaging-eventgrid/src/test/java/com/azure/messaging/eventgrid/EventGridPublisherClientTests.java

```diff
@@ -75,7 +75,7 @@ protected void beforeTest() {
     public void publishEventGridEvents() {
         EventGridPublisherAsyncClient<EventGridEvent> egClient = builder
             .endpoint(getEndpoint(EVENTGRID_ENDPOINT))
-            .credential(getKey(EVENTGRID_KEY))
+            .credential(TestUtil.getTestTokenCredential(interceptorManager))
```

Comment:
Is it permitted to leave the key-based tests as live-only? Switching to bearer auth means we're missing on the `AzureKeyCredential` coverage.